docs/template_plugin/tests/functional/op_reference/gather_nd.cpp

```diff
@@ -203,6 +203,188 @@ std::vector<GatherNDParams> generateCombinedParams() {
+                    Tensor(element::i32, {2, 2, 2}, std::vector<int32_t>{0, 1, 1, 0, 0, 0, 1, 1}),
+                    0,
+                    Tensor(IN_ET, {2, 2, 2}, std::vector<T>{12, 13, 20, 21, 10, 11, 22, 23}),
+                    "gather_nd_8_batch_1d_from_3d"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 2, 2}, std::vector<T>{10, 11, 12, 13, 20, 21, 22, 23}),
+                    Tensor(element::i32, {2, 2, 2}, std::vector<int32_t>{0, -1, -1, 0, 0, 0, 1, 1}),
+                    0,
+                    Tensor(IN_ET, {2, 2, 2}, std::vector<T>{12, 13, 20, 21, 10, 11, 22, 23}),
+                    "gather_nd_8_batch_1d_from_3d_negative"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 2, 2}, std::vector<T>{10, 11, 12, 13, 20, 21, 22, 23}),
+                    Tensor(element::i32, {2, 1, 1}, std::vector<int32_t>{1, 0}),
+                    0,
+                    Tensor(IN_ET, {2, 1, 2, 2}, std::vector<T>{20, 21, 22, 23, 10, 11, 12, 13}),
+                    "gather_nd_8_batch_2d_from_3d"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 3, 4}, std::vector<T>{1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+                                                            13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}),
+                    Tensor(element::i32, {2, 1}, std::vector<int32_t>{1, 0}),
+                    1,
+                    Tensor(IN_ET, {2, 4}, std::vector<T>{5, 6, 7, 8, 13, 14, 15, 16}),
+                    "gather_nd_8_batch_dims1"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 3, 4, 2}, std::vector<T>{
+                            1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15, 16,
+                            17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32,
+                            33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48}),
+                    Tensor(element::i32, {2, 3, 3, 2}, std::vector<int32_t>{
+                            1, 0, 3, 1, 2, 1, 0, 1, 1, 1, 2, 0, 3, 0, 3, 1, 2, 1,
+                            2, 0, 1, 1, 3, 1, 1, 1, 2, 0, 2, 0, 0, 0, 3, 1, 3, 1}),
+                    2,
+                    Tensor(IN_ET, {2, 3, 3}, std::vector<T>{
+                            3, 8, 6, 10, 12, 13, 23, 24, 22, 29, 28, 32, 36, 37, 37, 41, 48, 48}),
+                    "gather_8_nd_batch_dims2"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 3, 4}, std::vector<T>{1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+                                                            13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}),
+                    Tensor(element::i32, {2, 3, 1, 1}, std::vector<int32_t>{1, 0, 2, 0, 2, 2}),
+                    2,
+                    Tensor(IN_ET, {2, 3, 1}, std::vector<T>{2, 5, 11, 13, 19, 23}),
+                    "gather_8_nd_batch_dims2_lead_dims"),
```

Comment:
One more test case it will be very nice to have is `batch_dim=2, data_shape_{2, 3, 4, 5}, indice_shape={2,3, 2, 1} -> output_shape={2,3,2,5}`. In this case you gather slices of shape [5] instead of scalar

Reply:
Test was added